[hipDNN] Add pre-commits (#2132)
ROCm/rocm-libraries: CONTRIBUTING.md

However, once trunk-based development is in place, feature branches will be created directly from the default branch, `develop`.
During this period, a high priority will be placed on keeping the `develop` branch healthy.

## Pre-commit Hooks

Pre-commit hooks automatically run code quality checks before you commit changes, catching issues early in the development process. This includes formatting checks, linting, and other automated validations. See [`.pre-commit-config.yaml`](.pre-commit-config.yaml) for specifics.

### Setting Up Pre-commit

1. Install pre-commit:
```bash
pip install pre-commit
```

2. Install the git hooks:
```bash
cd rocm-libraries
pre-commit install
```

> Once installed, the hooks will run automatically on `git commit`. If any checks fail, the commit will be blocked until you fix the issues.

Reviewer: ```suggestion
After they are installed, the hooks will run automatically for `git commit`. If any checks fail, the commit will be blocked until you fix the issues.
```
